jobs/dynatrace-oneagent/templates/pre-start.erb

```diff
@@ -107,9 +119,37 @@ setHostProps() {
     echo -n "${HOST_PROPS}" > "${hostPropsFile}"
 }
 
+validateAgent() {
+    installLog "Validating the downloaded file..." 
+    if [[ -e $1 ]]; then
```

Comment:
Nitpicking: if the file doesn't exist, I'd put this verification on downloadAgent, since it was the download that failed, where here we're verifying whether such file is correct.

Reply:
Added check to if in downloadAgent